src/bindings/node/node.cc

```diff
@@ -0,0 +1,175 @@
+#include <node_api.h>
+#include <assert.h>
+#include <sstream>
+#include <string>
+#include <iostream>
+#include <boost/make_shared.hpp>
+#include <boost/noncopyable.hpp>
+#include <boost/optional.hpp>
+#include <boost/property_tree/json_parser.hpp>
+#include <boost/property_tree/ptree.hpp>
+#include <boost/shared_ptr.hpp>
+
+#include "valhalla/tyr/actor.h"
+
+boost::property_tree::ptree json_to_pt(const std::string& json) {
+  std::stringstream ss;
+  ss << json;
+  boost::property_tree::ptree pt;
+  boost::property_tree::read_json(ss, pt);
+  return pt;
+}
+
+boost::property_tree::ptree make_conf(const std::string config) {
+  return json_to_pt(config);
+}
+
+// check if napi_status is nap_ok, throw error if not
+void checkNapiStatus(napi_status status, napi_env env, const char* error_message) {
+  if (status != napi_ok) {
+    napi_throw_error(env, NULL, error_message);
+  }
+}
+
+#define DECLARE_NAPI_METHOD(name, func)                          \
+  { name, 0, func, 0, 0, 0, napi_default, 0 }
+
+class Actor {
+ public:
+  static napi_value Init(napi_env env, napi_value exports) {
+    napi_status status;
+    napi_property_descriptor properties[] = {
+        DECLARE_NAPI_METHOD("route", Route),
+    };
+
+    napi_value cons;
+    status =
+        napi_define_class(env, "Actor", NAPI_AUTO_LENGTH, New, nullptr, 1, properties, &cons);
+    assert(status == napi_ok);
+
+    status = napi_create_reference(env, cons, 1, &constructor);
+    assert(status == napi_ok);
+
+    status = napi_set_named_property(env, exports, "Actor", cons);
+    assert(status == napi_ok);
+    return exports;
+  }
+  static void Destructor(napi_env env, void* nativeObject, void* /*finalize_hint*/) {
+    reinterpret_cast<Actor*>(nativeObject)->~Actor();
+  }
+
+ private:
+  explicit Actor(std::string config)
+      : actor(make_conf(config)), env_(nullptr), wrapper_(nullptr) {}
+  ~Actor() { napi_delete_reference(env_, wrapper_); }
+
+  static napi_value New(napi_env env, napi_callback_info info) {
+    napi_status status;
+
+    napi_value target;
+    status = napi_get_new_target(env, info, &target);
+    assert(status == napi_ok);
+    bool is_constructor = target != nullptr;
+
+    if (is_constructor) {
+      // Invoked as constructor: `new Actor(...)`
+      size_t argc = 1;
+      napi_value jsthis;
+      napi_value argv[1];
+
+      status = napi_get_cb_info(env, info, &argc, argv, &jsthis, NULL);
+      checkNapiStatus(status, env, "Failed to parse arguments");
+
+      char buffer[5000];
+      size_t buffer_size;
+
+      // parse config string passed in
+      status = napi_get_value_string_utf8(env, argv[0], buffer, 5000, &buffer_size);
+      checkNapiStatus(status, env, "Failed to parse config to string");
+
+      const std::string parsed_config_string(buffer, buffer_size);
+      Actor* obj = new Actor(parsed_config_string);
```

Comment:
There should be a corresponding `delete ` somewhere, probably in `Destructor` - otherwise, we might leak these objects.